classes/class04/hw4/app.js

```diff
@@ -0,0 +1,42 @@
+//References: http://jsbin.com/igoVoTEr/1/edit?html,css,js,output
+
+var express = require("express");
+var path = require("path");
+var logger = require("morgan");
+var cookieParser = require("cookie-parser");
+var bodyParser = require("body-parser");
+var exphbs  = require("express-handlebars");
+var mongoose = require("mongoose");
+
+var ingredientRoutes = require("./routes/ingredients");
+// var getCat = require("./routes/getCat");
+
+var app = express();
+
+var PORT = 3000;
```

Comment:
Sometimes heroku (or another hosting provider) will want to be able to select which port your app uses with an environment variable so they can manage it along with all the other apps they host. Because of this, we tend to prefer to use
```
var PORT = process.env.PORT || 3000;
```
which will first check if an environment variable called `PORT` has been set, and if not, then use the value `3000` to set `var PORT`. By hardcoding `3000` you take away heroku's ability to choose the port number and run the risk of not being able to use port 3000 on their servers, which means your app will end up unavailable or hitting some connection error.